model/modelindexer/indexer.go

```diff
@@ -362,12 +362,19 @@ func (i *Indexer) flush(ctx context.Context, bulkIndexer *bulkIndexer) error {
 
 	resp, err := bulkIndexer.Flush(ctx)
 	if err != nil {
-		atomic.AddInt64(&i.eventsFailed, int64(n))
 		logger.With(logp.Error(err)).Error("bulk indexing request failed")
 		if tx != nil {
 			tx.Outcome = "failure"
 			apm.CaptureError(ctx, err).Send()
 		}
+
+		var errTooMany *errorTooManyRequests
+		// 429 may be returned as errors from the bulk indexer.
+		if errors.As(err, &errTooMany) {
+			atomic.AddInt64(&i.tooManyRequests, int64(n))
+		} else {
+			atomic.AddInt64(&i.eventsFailed, int64(n))
```

Comment:
I interpreted the `eventsFailed` counter to contain all failures. ES sending a `429` is a failure as the events are not indexed. I think it adds value to record these failures in more detail, but I'd see the `tooManyRequests` as subcategory of `eventsFailed` and therefore expected to increase the counter for both.